backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/projection/impl/ElasticsearchDistanceToFieldProjection.java

```diff
@@ -36,7 +39,8 @@
 
 	private final String scriptFieldName;
 
-	ElasticsearchDistanceToFieldProjection(String absoluteFieldPath, GeoPoint center, DistanceUnit unit) {
+	ElasticsearchDistanceToFieldProjection(Set<URLEncodedString> indexNames, String absoluteFieldPath, GeoPoint center, DistanceUnit unit) {
```

Comment:
Wouldn't an abstract class make sense here? You could even turn `ElasticsearchSearchProjection` into an abstract class.

Reply:
When I wrote it I was thinking to do so. However then I asked to myself: what can we share among all `ElasticsearchSearchProjection`s?
I think that extract an abstract class just for the field `Set<String> indexNames;` is too much.

Reply:
Just to be clear, I wasn't talking about just Elasticsearch, I think it would make sense for Lucene too.

I feel like turning `ElasticsearchSearchProjection` wouldn't be such a big change, especially considering all the other changes you had to make and considering that it would bring projections closer to `ElasticsearchSearchPredicate` and `ElasticsearchSearchSort`.

But I can understand your point of view too. It can stay that way, it's not a big deal.